Fix condition mapping for payee rule creation from payee modal

winklevos opened the pull request:
Fixes [bug](https://github.com/actualbudget/actual/issues/195) where creating rule from Payee modal does not auto populate the condition with the payee. 

**Before**
![image](https://user-images.githubusercontent.com/30995408/210022994-385811f5-9876-4a68-b2d8-712d5574cb5d.png)


**After**
![image](https://user-images.githubusercontent.com/30995408/210023006-2abd2992-4012-4e73-81c8-e69fb4ad3f98.png)

winklevos commented:
@MatissJanis PR checks need approval if you could please

MatissJanis commented:
👋 Hi,

Apologies for the slow turn-around time. There is still a small issue. Here's a demo:
![Kapture 2023-01-05 at 18 41 46](https://user-images.githubusercontent.com/886567/210856413-dd02313e-4e90-4fad-b311-4085fe3be55c.gif)

Basically if a rule for payee already exists AND you try to create another: the subsequent modal window is not pre-filled with the payee either. If we merge your fix, this might create confusion as the experience will be different between the two states.

Would you mind taking a look at this too? Then I'll be happy to merge.

Thanks!

winklevos commented:
@MatissJanis 

I fixed up the additional create rule context, aligned the new default role to have an action for category with a null placeholder. 

Also removed an extra lookup, seems to work fine without it and can't find any use for it

MatissJanis commented on packages/desktop-client/src/components/ManageRules.js:
```diff
@@ -588,20 +588,34 @@ export default function ManageRules({
+        {
+          field: 'payee',
+          op: 'is',
+          value: null,
```
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏could you please move the expression here? It will improve the readability.
```suggestion
          value: payeeId || null,
```

MatissJanis commented on packages/desktop-client/src/components/ManageRules.js:
```diff
@@ -588,20 +588,34 @@ export default function ManageRules({
 
   function onCreateRule() {
+    let rule = {
+      id: null,
```
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why do we need to set `id` here? What happens if we do not set it?

winklevos replied:
I first tried to remove `id: null and stage:null` which threw on save, but that is related to stage. I have removed as per the comment

MatissJanis commented on packages/desktop-client/src/components/ManageRules.js:
```diff
@@ -588,20 +588,29 @@ export default function ManageRules({
+        {
+          op: 'set',
+          field: 'category',
+          value: payeeId || null,
```
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏is this correct? `payeeId` being set on the `category` field.

winklevos replied:
Apologies, pre-coffee commit. Fixed

MatissJanis reviewed (approved):
Did a bit of testing and not finding any issues. IMO looks good.

Thanks for your contribution!
